docs/tutorial/reading_writing.md

```diff
@@ -0,0 +1,336 @@
+
+@:select(api-style)
+
+@:choice(syntax)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.syntax.path.*
+import fs2.io.file.Path
+import cats.effect.{IO, IOApp}
+
+object App extends IOApp.Simple:
+
+  val path = Path("testdata/change_me.txt")
+
+  def run: IO[Unit] =
+    for
+      file <- path.read
+      reversedFile = file.reverse
+      _ <- path.write(reversedFile)
+    yield ()
+
+end App
+```
+
+@:choice(static)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.FilesOs
+import fs2.io.file.Path
+import cats.effect.{IO, IOApp}
+
+object App extends IOApp.Simple:
+
+  val path = Path("testdata/change_me.txt")
+
+  def run: IO[Unit] =
+    for
+      file <- FilesOs.read(path)
+      reversedFile = file.reverse
+      _ <- FilesOs.write(path, reversedFile)
+    yield ()
+
+end App
+```
+
+@:@
+
+Be aware that this will overwrite the contents of the file. So be careful not to change important files while you are learning!
+
+### Exercise
+
+Try loading the contents of two different files, concatenating them, and saving the result to a third location. How would you do it?
+
+[See possible solution](https://gist.github.com/99880af9b051b15f8adace7fe7aee2e3.git)
+
+## Working line by line
+
+There is a useful variant of the standalone methods called `Lines` (e.g. `readLines`), which reads the file line by line and stores them on a `List[String]`. This comes handy when you are working with a list of things that you want to convert:
+
+`testdata/names.data`
+
+```
+Alex
+Jamie
+Morgan
+Riley
+Taylor
+Casey
+River
+```
+
+@:select(api-style)
+
+@:choice(syntax)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.syntax.path.*
+import fs2.io.file.Path
+import cats.effect.{IO, IOApp}
+
+object Names extends IOApp.Simple:
+
+  case class Name(value: String)
+
+  val namesPath = Path("testdata/names.data")
+
+  def run: IO[Unit] =
+    for
+      lines <- namesPath.readLines      // (1)
+      names = lines.map(Name(_))        // (2)
+      _ <- IO(names.foreach(println))   // (3)
+    yield ()
+
+end Names
+```
+
+@:choice(static)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.FilesOs
+import fs2.io.file.Path
+import cats.effect.{IO, IOApp}
+
+object Names extends IOApp.Simple:
+
+  case class Name(value: String)
+
+  val namesPath = Path("testdata/names.data")
+
+  def run: IO[Unit] =
+    for
+      lines <- FilesOs.readLines(namesPath)   // (1)
+      names = lines.map(Name(_))              // (2)
+      _ <- IO(names.foreach(println))         // (3)
+    yield ()
+
+end Names
+```
+
+@:@
+
+In the example from above, we first load the list of names as a `List` of `String` (1), then we map over this list and convert the strings to a `List` of `Name` (2), and finally we print each of one the elements recently mapped to the console (3). Note that using `println` with `foreach` creates side effects, that's why we wrap the computation on an `IO`.
```

Comment:
So when you have numbered items like this you'll want to structure the footnotes like:
```
(1) load the list of names as `List`
(2) convert to `Name`
(3) print
```
This is easier to parse (imo!!) for most people. However I think this is a nit, and very subjective so feel free to completely ignore this lol

Reply:
+1